### PR TITLE
Make left panel remote fetch use the same function as FormBrowse

### LINKED ISSUE #6047: Fetching orgin from the remote branches on first load asks for the ssh key
## Current behaviour
Trying to fetch the origin from the branches view asks to load the ssh key
(No supported authentication methods available (server sent: public key)

It seems that the key is not loaded, while fetching first the master from the commit list will load the key correctly making the following fetch from the remote tree to work

## Expected behaviour
The branch should be fetched.

## Steps to reproduce
1. close Git Extensions and Pageant
2. open a repository which uses ssh
3. fetch the origin from the remote branches tree on the left

## Screenshots
Image

### PR DESCRIPTION
**Pull request by amaiorano**
This ensures that Pageant is started automatically, if desired, etc.

Fixes #6047

## Test methodology 

- Generated SSH key with Putty, added to my GitHub, etc.
- Setup my remote to use SSH, and to auto-launch Pageant with private key file
- Closed Pageant, then fetched using the left panel remote node, making sure it works like when fetching from the toolbar.

## Test environment(s) 

- GIT 2.19.1.windows.1
- Windows 10

### CODE REVIEW

**Comment by vbjay** [comment]
Verify these are all good.

image

**Comment by amaiorano** [comment]
> Verify these are all good.
> 
> image

Good call, @vbjay. I had missed the Fetch operation for remote branches. Will update PR soon.

**Comment by amaiorano** [comment]
Okay, the change got a little bigger now. As the second commit comment says:

```
Refactor left panel remote branch fetch, create, delete, checkout, merge, and rebase to use UICommands

Also made it so that FetchAnd* commands don't perform the second action if the fetch fails or is cancelled.
```

There is still one command in there that does its own form thing, and that's Reset (remote branch). I added a TODO in there:

```csharp
            public void Reset()
            {
                // TODO: Move this into new function UICommands.StartResetCurrentBranchDialog

                var objectId = Module.RevParse(FullPath);

                if (objectId == null)
                {
                    MessageBox.Show($"Branch \"{FullPath}\" could not be resolved.");
                }
                else
                {
                    using (var form = new FormResetCurrentBranch(UICommands, Module.GetRevision(objectId)))
                    {
                        form.ShowDialog(TreeViewNode.TreeView);
                    }
                }
            }
```

Presently, there are three places that create their own FormResetCurrentBranch: in FormReflog.cs, RevisionGridControl.cs, and of course, RepoObjectsTree.Nodes.Remotes.cs. We should refactor all three in terms of a UICommands.StartResetCurrentBranchDialog. I think we should do this in a separate change.

**Comment by amaiorano** [comment]
Would someone like to review this change?

File: GitUI/BranchTreePanel/RepoObjectsTree.Nodes.Remotes.cs [hunk 1]
```diff
@@ -193,6 +182,46 @@ public void Reset()
...
 
+            public bool FetchAndMerge()
+            {
+                if (Fetch())
+                {
+                    return Merge();
+                }
+
+                return false;
+            }
```
Comment (RussKie) [inline GitUI/BranchTreePanel/RepoObjectsTree.Nodes.Remotes.cs]: Can this be simplified to `return Fetch() && Merge()`?

Comment (amaiorano) [reply GitUI/BranchTreePanel/RepoObjectsTree.Nodes.Remotes.cs]: Absolutely! Haha, I was actually planning on doing that, which is why I added bool return values everywhere, but forgot. Thanks :)

Comment (amaiorano) [reply GitUI/BranchTreePanel/RepoObjectsTree.Nodes.Remotes.cs]: Amended last commit with these changes.

**Pull request merged**